UI/MemStickScreen.cpp

```diff
@@ -51,49 +112,91 @@ void MemStickScreen::CreateViews() {
 
 	Margins actionMenuMargins(15, 15, 15, 0);
 
-	root_ = new AnchorLayout();
+	root_ = new LinearLayout(ORIENT_HORIZONTAL);
 
-	ViewGroup *columns = new LinearLayout(ORIENT_HORIZONTAL, new AnchorLayoutParams(FILL_PARENT, FILL_PARENT));
-	root_->Add(columns);
+	Spacer *spacerColumn = new Spacer(new LinearLayoutParams(20.0, FILL_PARENT, 0.0f));
+	ScrollView *leftColumnScroll = new ScrollView(ORIENT_VERTICAL, new LinearLayoutParams(1.0));
 
-	ViewGroup *leftColumn = new AnchorLayout(new LinearLayoutParams(1.0));
+	ViewGroup *leftColumn = new LinearLayoutList(ORIENT_VERTICAL);
 	ViewGroup *rightColumnItems = new LinearLayout(ORIENT_VERTICAL, new LinearLayoutParams(300, FILL_PARENT, actionMenuMargins));
-	columns->Add(leftColumn);
-	columns->Add(rightColumnItems);
+	root_->Add(spacerColumn);
+	root_->Add(leftColumnScroll);
+	root_->Add(rightColumnItems);
+
+	leftColumnScroll->Add(leftColumn);
+
+	if (initialSetup_) {
+		leftColumn->Add(new TextView(iz->T("Welcome to PPSSPP!"), ALIGN_LEFT, false));
+		leftColumn->Add(new Spacer(new LinearLayoutParams(FILL_PARENT, 12.0f, 0.0f)));
+	}
 
-	Path path = g_Config.memStickDirectory;
-	int64_t freeSpaceAtMemStick = -1;
+	if (System_GetPropertyBool(SYSPROP_ANDROID_SCOPED_STORAGE)) {
+		leftColumn->Add(new TextView(iz->T("ScopedStorageWarning", "WARNING: BETA ANDROID SCOPED STORAGE SUPPORT\nMAY EAT YOUR DATA"), ALIGN_LEFT, false));
+	}
+
+	leftColumn->Add(new TextView(iz->T("MemoryStickDescription", "Choose PSP data storage (Memory Stick)"), ALIGN_LEFT, false));
+
+	// For legacy Android systems, so you can switch back to the old ways if you move to SD or something.
+	// TODO: Gonna need a scroll view.
 #if PPSSPP_PLATFORM(ANDROID)
-	if (Android_IsContentUri(path.ToString())) {
-		freeSpaceAtMemStick = Android_GetFreeSpaceByContentUri(path.ToString());
-	} else {
-		freeSpaceAtMemStick = Android_GetFreeSpaceByFilePath(path.ToString());
+	if (!System_GetPropertyBool(SYSPROP_ANDROID_SCOPED_STORAGE)) {
+		leftColumn->Add(new Choice(iz->T("Use PSP folder at root of storage")))->OnClick.Handle(this, &MemStickScreen::OnUseStorageRoot);
+		leftColumn->Add(new TextView(iz->T("ChooseFolderDesc", "* Data will stay even if you uninstall PPSSPP.\n* Data can be shared with PPSSPP Gold\n* Easy USB access"), ALIGN_LEFT, false));
```

Comment:
`Data can be shared with PPSSPP Gold` doesn't work really well for Gold version itself, maybe something like: "Data can be shared between normal and Gold version of PPSSPP".

The path `Android/data/org.ppsspp.ppsspp/files` displayed below should be different for the gold version too to avoid confusion.

Reply:
Oh yeah, I noticed this a few days ago and then completely forgot to fix it. Thanks!